WeatherApp/Model/WeatherResponse.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  WeatherResponse.swift
+//  WeatherApp
+//
+//  Created on 25/01/2020.
+//  Copyright © 2020 Marcin Wójciak. All rights reserved.
+//
+
+import Foundation
+
+struct WeatherResponse: Decodable {
+    enum CodingKeys: String, CodingKey {
+        case city
+        case dayForecasts = "list"
+    }
+
+    let city: CityInfoResponse
+    let dayForecasts: [DayForecastResponse]
+
+    init(from decoder: Decoder) throws {
+        do {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+            city = try container.decode(CityInfoResponse.self, forKey: .city)
+            dayForecasts = try container.decode([DayForecastResponse].self, forKey: .dayForecasts)
+        } catch let error {
+            fatalError("Could not decode response.\nError: \(error)")
+        }
+    }
+}
+
+struct CityInfoResponse: Decodable {
```

Comment:
Tutaj mamy drugi przykład implementowania protokołu `Decodable`. Gdy wszystkie nazwy oraz tyoy naszych zmiennych pokrywają się z kluczami z API i typami ich wartości musimy dopisać tylko `: Decodable` do naszego typu. Nie musimy robić `enum CodingKeys: String, CodingKey { ... }` ani `init(from decoder: Decoder)`. Wszystko dzieje się samo - magic :P Jeśli API jest dobrze skonstruowane, to zazwyczaj na takim przypadku kończy się mapowanie ;)